Source/Evolve/WarpXEvolve.cpp

```diff
@@ -469,14 +463,11 @@ WarpX::OneStep_nosub (Real cur_time)
             amrex::Abort(" Medium for EM is unknown \n");
         }
 
-        FillBoundaryE(guard_cells.ng_FieldSolver);
+        FillBoundaryE(guard_cells.ng_FieldSolver, WarpX::sync_nodal_points);
         EvolveF(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveG(0.5_rt * dt[0], DtType::SecondHalf);
         EvolveB(0.5_rt * dt[0], DtType::SecondHalf); // We now have B^{n+1}
-
-        // Synchronize E and B fields on nodal points
-        NodalSync(Efield_fp, Efield_cp);
-        NodalSync(Bfield_fp, Bfield_cp);
+        FillBoundaryB(guard_cells.ng_FieldSolver, WarpX::sync_nodal_points);
```

Comment:
This call to `FillBoundaryB` was added in order to keep the nodal synchronization otherwise removed when deleting the call to `NodalSync`.

Reply:
Note that calling `MultiFab::OverrideSync` is now equivalent to calling `FillBoundaryAndSync` with zero ghost cells.  So if the only purpose here is to synchronize, not to fill ghost cells, `NodalSync` or `FillBoundaryB` with zero ghost cells should be used.

Reply:
OK, thank you, then I think I'll keep the call to `NodalSync` for now. In my opinion it does make sense to also fill the guard cells at this point, as B has been updated, but I think it would make sense to do this while we also maybe remove other redundant calls to `FillBoundarB` (without synchronization) that appear in other parts of the PIC loop. This is part of the optimizations mentioned in #3027, so it's probably to try them separately in small PRs. Here I'll try to just merge the calls when possible, without changing the number of calls to fill the guard cells or synchronize.